### ISSUE TITLE
Error: annotation "app.openshift.io/vcs-uri"=git@xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx" value length exceeds Kubernetes max 63

### ISSUE THREAD

**Issue by jeffgus** [redhat-developer/odo#7237]
/kind bug

## What versions of software are you using?

Fedora 39

odo v3.15.0 (10b5e8a8f)

## How did you run odo exactly?

odo dev --platform podman

## Actual behavior
```
Error occurred on Push - exit status 125: 
Complete Podman output:
Error: annotation "app.openshift.io/vcs-uri"="xxxxxxxxx" value length exceeds Kubernetes max 63
```

odo dev used against Openshift seems to work. It doesn't complain about any annotation.

## Expected behavior
Pods should start.

## Any logs, error output, etc?
```
Error occurred on Push - exit status 125: 
Complete Podman output:
Error: annotation "app.openshift.io/vcs-uri"="xxxxxxxxx" value length exceeds Kubernetes max 63
```

**Comment by jeffgus** [comment]
If I change the repo to a shorter string in the .git/config, then pod will start using `odo dev --platform podman`

**Comment by jeffgus** [comment]
I installed podman 5 from (currentlly pre-release) Fedora 40 repo. The new podman does NOT trigger this error. Should this ticket be closed if the problem only happens on pre-podman 5 releases?

**Comment by rm3l** [comment]
> I installed podman 5 from (currentlly pre-release) Fedora 40 repo. The new podman does NOT trigger this error. Should this ticket be closed if the problem only happens on pre-podman 5 releases?

Thanks for reporting this issue, @jeffgus ! Indeed, I found this related issue (https://github.com/containers/podman/issues/21663), which has been fixed in Podman [v5.0.0](https://github.com/containers/podman/releases/tag/v5.0.0).

So the recommendation would be to use Podman v5+. Or, if not possible, as a temporary workaround, rename the Git remote into anything other than `origin` (so that `odo` won't try to add the annotation).

Closing this issue then, but feel free to reopen if you think otherwise.

/close not-planned
